fix(github): fix zerotime issues for GraphQL

petkostas opened the pull request:
### Summary
GitHub GraphQL returns zero time (`&time.Time{}`) for `nullable` fields, this leads to GORM not properly handling this cases.
This PR:
- Introduces a new core helper util `NilIfZeroTime` in order to ensure that `*time.Time` fields are assigned the `nil` value.
- Fixes the relevant cases in both collectors and extractors (as existing entries in the DB should be properly handled as well).


### Does this close any open issues?
Closes #8538 

### Screenshots
⚠️  Screenshots pending, as this PR has not been tested yet.

⚠️  This PR has not been tested yet, I am looking for cases to properly test the fixes ⚠️

petkostas committed:
fix(github): fix zerotime issues for GraphQL

klesh commented:
Hi @petkostas, I saw the review request this morning, but since the PR is still in draft, I wasn’t sure if it’s ready for review yet?

petkostas commented:
Hey @klesh 
I need to test this potentially with repo that re-produces the initial problem. I added you in case you wanted to do a first check regarding the approach.
But feel free to ignore until I have tested it and removed the draft.

petkostas committed:
Merge branch 'main' into fix/issue-8538-fix-github-zero-time

petkostas committed:
Merge branch 'apache:main' into fix/issue-8538-fix-github-zero-time

Commit:
fix(lint): fix linting issue

petkostas commented:
Opened a new directly to repo:
https://github.com/apache/incubator-devlake/pull/8583
